### ISSUE TITLE
False positive ShellInjection

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#387]
### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - shell: cp {0} x.json
           run: ${{ toJSON(steps.check-file-changes.outputs) }}
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

No finding, there's no injection problem if there's no string interpolation happening.

### Actual behavior

`ShellScriptInjection` flagged

### Additional Context

a8120
